src/core.ts

```diff
@@ -557,11 +583,17 @@ function TrackerCore(schemas: Schemas, sanitizeEvent: SanitizeEventFn): TrackerF
                     throw new Error('call autoSetVisitor() at least once before an event is sent');
                 }
 
-                (event as FullSendableEvent).clientVersion = { raw: internals.VERSION };
-                (event as FullSendableEvent).eventType = type;
-                (event as FullSendableEvent).customer = internals.CUSTOMER;
-                (event as FullSendableEvent).visit = internals.VISIT as SendableVisit;
-                return event as FullSendableEvent;
+                const metadata = internals.getPreparedMetadata(event.metadata);
+                const fullSendableEvent: FullSendableEvent = {
+                    ...event,
+                    clientVersion: { raw: internals.VERSION },
+                    eventType: type,
+                    customer: internals.CUSTOMER,
+                    visit: internals.VISIT as SendableVisit,
+                    ...metadata && { metadata }
```

Comment:
This is kind of misleading at a glace since `...` has lower precedence than `&&`, but the spacing suggests otherwise. I would use parentheses:
```suggestion
                    ...(metadata && { metadata })
```